src/main/java/net/coderbot/iris/gui/element/ShaderPackListWidget.java

```diff
@@ -30,30 +37,36 @@ public ShaderPackListWidget(MinecraftClient client, int width, int height, int t
 
 	@Override
 	public int getRowWidth() {
-		// Allow the list widget to expand to take up most of the width of the screen, or shrink as needed.
-		// This is important both because shader pack names can be quite long, and it also helps if this list widget
-		// is side-by-side with another widget, such as a config GUI.
-		return width - 50;
+		// Temporarily set to only reach a width of up to 312 in order to fit in with
+		// the width of the array of buttons at the bottom of the GUI. May be changed
+		// in the future if this widget is made to occupy half the screen.
+		return Math.min(308, width - 50);
 	}
 
 	@Override
 	protected int getRowTop(int index) {
 		return super.getRowTop(index) + 2;
 	}
 
+	private static boolean isValidPackPath(Path path) {
+		return !path.equals(Iris.SHADERPACKS_DIRECTORY) && Iris.isValidShaderpack(path);
+	}
+
 	public void refresh() {
 		this.clearEntries();
 
 		try {
+			this.addEntry(enableShadersButton);
+
 			Path path = Iris.SHADERPACKS_DIRECTORY;
-			int index = -1;
+			int index = 0;
 
 			for (String pack : BUILTIN_PACKS) {
 				index++;
 				addEntry(index, pack);
 			}
 
-			Collection<Path> folders = Files.walk(path, 1).filter(Iris::isValidShaderpack).collect(Collectors.toList());
+			Collection<Path> folders = Files.walk(path, 1).filter(ShaderPackListWidget::isValidPackPath).collect(Collectors.toList());
```

Comment:
Perhaps try https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#list-java.nio.file.Path- instead? Seems like it would solve your problem in a nicer way:

> Some file systems maintain special links to the directory itself and the directory's parent directory. Entries representing these links are not included.

This is also probably relevant in the code for locating the `shaders` directory within a shaderpack.

Reply:
Will be taken care of.